### PR TITLE
Identity from path

### PR DESCRIPTION
**Pull request by ErikDeSmedt**
Implements `TlsConfig::identity_from_path`.

This is a utility method that reads an identity (cert + key) from a path.

The python implementation might be a bit controversial. I have removed the `id` which is a breaking change.

The id is the combination of `cert` + `key` and is only used in the python version. When using `identity_from_path` there are no python bindings available to request the `cert` + `key` and construct the `id` gain.

The alternative is to read the `client.crt` and `client-key.pem` file in python and use the standard `tls.identity`-call. However, I prefer not to duplicate the code.

### CODE REVIEW

**Review by nepet — APPROVED** [review]
ACK 709a76413e409b7a4a14b78777f6df398b695400

Was too fast! Seems that gl-testing relies on the `id` field!

File: libs/gl-client-py/glclient/tls.py [hunk 1]
```diff
@@ -11,7 +11,6 @@ def __init__(self):
         # through the rust library (streaming calls)
         self.inner = native.TlsConfig()
         self.ca: Optional[bytes] = None
-        self.id: Tuple[Optional[bytes], Optional[bytes]] = (None, None)
```
Comment (nepet) [inline libs/gl-client-py/glclient/tls.py:14]: This id field is accessed by some tests from `gltesting/tests/test_gl_node.py` They are necessary there in order to set up a custom gprc connection. I thing the custom grpc connection is not needed anymore and actually not the right way to access greenlight.
I can add a fix on top of your commits if you like, as I wrote the tests that need the id field in the first place.

Comment (ErikDeSmedt) [reply libs/gl-client-py/glclient/tls.py:14]: That would be great

**Review by nepet — APPROVED** [review]
Added the test fixes on top and rebased on main

**Pull request merged**